pkg/istio-agent/sds-agent.go

```diff
@@ -264,6 +268,40 @@ func gatewaySdsExists() bool {
 	return !os.IsNotExist(err)
 }
 
+// explicit code to determine the root CA to be configured in bootstrap file.
+// It may be different from the CA for the cert server - which is based on CA_ADDR
+// Replaces logic in the template:
+//                 {{- if .provisioned_cert }}
+//                  "filename": "{{(printf "%s%s" .provisioned_cert "/root-cert.pem") }}"
+//                  {{- else if eq .pilot_cert_provider "kubernetes" }}
+//                  "filename": "./var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+//                  {{- else if eq .pilot_cert_provider "istiod" }}
+//                  "filename": "./var/run/secrets/istio/root-cert.pem"
+//                  {{- end }}
+//
+// In addition it deals with the case the XDS server is on port 443, expected with a proper cert.
+// /etc/ssl/certs/ca-certificates.crt
+//
+// TODO: additional checks for existence. Fail early, instead of obscure envoy errors.
+func (sa *Agent) FindRootCAForXDS() string {
+	if strings.HasSuffix(sa.proxyConfig.DiscoveryAddress, ":443") {
+		return "/etc/ssl/certs/ca-certificates.crt"
```

Comment:
This is not a safe assumption. Should we use golang code for this? https://golang.org/src/crypto/x509/root_linux.go

Reply:
We could do this - but for sidecar we control the image. VMs are different story.

Reply:
Added a comment - the file is private, and we use that path in other places.

I think we can solve this when we add support for other platforms - I know we kind of support RedHat, but
 we can copy the file to this location as well in installer - it's too much craziness otherwise.